### PR TITLE
Adding the useCrossRegionInference Setting for Amazon Bedrock

### PR DESCRIPTION
**Pull request by philikai**
Updating the Bedrock section as well as the frontend to accommodate for using cross-region inference requests. 
Cross-region inference requests double the throughput limits on Amazon Bedrock in US and EU regions. This was implemented as requested on issue:
- https://github.com/saoudrizwan/claude-dev/issues/229
- https://github.com/saoudrizwan/claude-dev/issues/279

To change the x-region request, only minimal changes needed on the model id, by prefixing "us." or "eu."
_<non cross-region inference>
"modelId":"anthropic.claude-3-5-sonnet-20240620-v1:0 "
</non cross-region inference>
vs
<cross-region inference>
"modelId":"**us**.anthropic.claude-3-5-sonnet-20240620-v1:0 "
</cross-region inference>_

Modified files:
- src/api/providers/bedrock.ts
- src/shared/api.ts
- webview-ui/src/components/settings/ApiOptions.tsx

Screenshot on new option setting:
image

### CODE REVIEW

File: src/api/providers/bedrock.ts [hunk 1]
```diff
@@ -3,7 +3,9 @@ import { Anthropic } from "@anthropic-ai/sdk"
 import { ApiHandler, ApiHandlerMessageResponse } from "../"
 import { ApiHandlerOptions, bedrockDefaultModelId, BedrockModelId, bedrockModels, ModelInfo } from "../../shared/api"
 
-// https://docs.anthropic.com/en/api/claude-on-amazon-bedrock
+// Hardcoded flag to enable cross-region inference
+const USE_CROSS_REGION_INFERENCE = true
```
Comment (philikai) [inline src/api/providers/bedrock.ts]: Artefact from testing. Should be removed

**Comment by saoudrizwan** [comment]
Thank you for putting this together, some users had mentioned this issue and I wasn't sure how to address it. Will get this merged ASAP

**Comment by philikai** [comment]
Fixed issue with hardcoded default  of "use cross region inference" in code. 

Fixed issue with non persistent flag status between tasks and window sessions.

**Comment by saoudrizwan** [comment]
Hey @philikai if you could pls update your PR to work with the latest, that would be much appreciated!

**Comment by philikai** [comment]
@saoudrizwan Updated the merge to be in sync with Cline latest version. 

it is ready to merge.

**Comment by philikai** [comment]
Just checking in. Anything else that needs to be done for integration?

**Comment by RobertPaulson90** [comment]
Any news on this? Can't use the new Claude 3.5 sonnet v2 until this is released.

> 400 Invocation of model ID anthropic.claude-3-5-sonnet-20241022-v2:0 with on-demand throughput isn’t supported. Retry your request with the ID or ARN of an inference profile that contains this model.

**Review by RobertPaulson90 — APPROVED** [review]

**Comment by usathyan** [comment]
When is this expected? can you please merge this to main or provide details on how to enable this?

**Comment by fn-alves** [comment]
Please update the PR and merge it into main.

**Comment by philikai** [comment]
Ready for merge. Would be great if we finally mange to merge it, otherwise we will diverge once again when time passes.

**Comment by usathyan** [comment]
please merge this to main. I have this running in debug mode and have tested it. it works great.

**Comment by saoudrizwan** [comment]
Hey @philikai there were too many conflicts so I had to implement this fix myself, can you please take a look and let me know if I didn't do something correctly? https://github.com/cline/cline/commit/ad29ff2a03ddfd94249f4901826a7f1b02692f57

I apologize for not merging this PR sooner, I tried working through the conflicts in your branch so your contribution shows up in the repo but there were too many divergences. 

@usathyan @fn-alves @RobertPaulson90 v2.1.4 now supports cross-region inference, please let me know if it works or if you're still running into any issues.

Thank you for being patient with me 🙏

**Comment by usathyan** [comment]
Thank you! Much appreciated

**Pull request closed without merging**